### DISCUSSION TITLE
Possiblity to mount pool at boot time

### DISCUSSION THREAD

**Discussion by derritter88** [openzfsonwindows/openzfs#341] [Q&A]
Is there any possiblity to mount a ZFS pool at pool time?
I tried several options at Windows task scheduler but that would mount my pool after files would already be needed by the system.

**Comment by lundman** [comment]
Unsure how that part works as of yet. win-btrfs seem able to, so eventually we will figure out what it is needed.

**Comment by derritter88** [comment]
Would be cool. In my humble opinion ZFS for Windows is pretty stable currently and could be used as an alternativ for Microsofts native NTFS filesystem

**Comment by guenther-alka** [comment]
What I found is that after a reboot the pool is not imported. To access you need to import not mount. You need a script that you can call via task on bootup that execute a zpool import to detect all available pools followed by a zpool import poolname. 

If pool does not change,, simply call a "zpool import poolname" or "zpool import -f poolname" to  ignore warnings on import.

**Comment by lundman** [comment]
As for just importing the pool (after) booting, we are collecting methods here:
https://openzfsonosx.org/wiki/Windows_import
- eventually it will be part of the installation, but for now it's in the cosmetic-TODO list.

As for "files needed at boot", and the idea of booting on top of a ZFS, that is still in the future-TODO list. ZFS Needs to be more feature rich before I can try it, I only added swap-on-file last month for example.

**Comment by derritter88** [comment]
Okay I see - I will keep an eye on the topic and would leave the request open as a user story if you don't mind

**Comment by lundman** [comment]
Didn't even know you could close discussions :)

**Comment by derritter88** [comment]
Yeah - it works like a bug report :-)
